### PR TITLE
Add duplicate set option to Replicator and Bard

### PR DESCRIPTION
**Pull request by jacksleight**
This PR adds a duplicate set option to the Replicator and Bard fieldtypes. Closes https://github.com/statamic/ideas/issues/756 and closes https://github.com/statamic/ideas/issues/81.

https://user-images.githubusercontent.com/126740/170833249-ae490b51-b3a7-4520-b8de-f579096d4774.mp4

I wasn't sure if there's a simple way for `BardFieldtype` to access the node attributes and position just using the set ID, which is why I'm parsing those values to `duplicateSet()` along with the ID.

### CODE REVIEW

File: resources/js/components/fieldtypes/replicator/Set.vue [hunk 1]
```diff
@@ -199,6 +204,10 @@ export default {
             this.$emit('expanded');
         },
 
+        duplicate() {
+            this.$emit('duplicate');
```
Comment (jasonvarga) [inline resources/js/components/fieldtypes/replicator/Set.vue:208]: Can you change this to `duplicated` for consistency?

File: resources/js/components/fieldtypes/replicator/Replicator.vue [hunk 2]
```diff
@@ -33,8 +33,10 @@
                     :field-path-prefix="fieldPathPrefix || handle"
                     :has-error="setHasError(index)"
                     :previews="previews[set._id]"
+                    :can-add-set="canAddSet"
                     @collapsed="collapseSet(set._id)"
                     @expanded="expandSet(set._id)"
+                    @duplicate="duplicateSet(set._id)"
```
Comment (jasonvarga) [inline resources/js/components/fieldtypes/replicator/Replicator.vue:39]: Can you change this to `@duplicated` for consistency?

**Review by jasonvarga — CHANGES_REQUESTED** [review]

**Comment by jasonvarga** [comment]
This is dope.

File: resources/js/components/fieldtypes/replicator/Set.vue (hunk 1 above)
Comment (jasonvarga) [reply resources/js/components/fieldtypes/replicator/Set.vue]: (I did it)

File: resources/js/components/fieldtypes/replicator/Replicator.vue (hunk 2 above)
Comment (jasonvarga) [reply resources/js/components/fieldtypes/replicator/Replicator.vue]: (I did it)

**Comment by Hesesses** [comment]
Awesome!

**Pull request merged**